wandb/sdk/interface/artifacts/artifact.py

```diff
@@ -81,6 +81,11 @@ def name(self) -> str:
         """The artifact's name."""
         raise NotImplementedError
 
+    @property
+    def full_name(self) -> str:
+        """The artifact's full name."""
+        raise NotImplementedError
```

Comment:
Let's implement this here and let the children inherit. Resolving the indirection in `wandb_run.py` will still be necessary, but then we won't need to touch `public.py` and `wandb_artifacts.py` will be a little simpler.
```suggestion
        return f"{self.entity}/{self.project}/{self.name}"
```